g3/lowpan6_g3.c

```diff
@@ -822,6 +832,26 @@ lowpan6_g3_status_handle(struct netif *netif, struct pbuf *p, struct lowpan6_lin
   }
 
   is_route_repair = loadng_g3_route_repair_status(netif, final_dest);
+
+  /* Blacklist table update */
+  if (status == g3plc_mac_status_no_ack) {
+	blacklisted = lowpan6_g3_blacklist_table_add(lowpan6_link_addr_to_u16(dest));
+	if (blacklisted == NULL) {
+		  LWIP_DEBUGF(LWIP_LOWPAN6_DEBUG,("lowpan6_g3_status_handle: failed to add %02X%02X to blacklist table\n",
+											dest->addr[0], dest->addr[1]));
+	}
+  } else if (status == g3plc_mac_status_success) {
+	blacklisted = lowpan6_g3_blacklist_table_lookup(lowpan6_link_addr_to_u16(dest));
+	if (blacklisted != NULL) {
+		lowpan6_g3_blacklist_table_remove(blacklisted);
+		LWIP_DEBUGF(LWIP_LOWPAN6_DEBUG,("lowpan6_g3_status_handle: device %02X%02X deleted from blacklist\n",
+										dest->addr[0], dest->addr[1]));
+	}
```

Comment:
The code in `loadng_g3.c` resets entry’s `valid_time` to `ctx->blacklist_table_ttl`. It probably doesn’t make any difference right now, but it’s better to handle this consistently across all places.